hphp/runtime/vm/jit/vasm-fold-imms.cpp

```diff
@@ -421,6 +421,12 @@ struct ImmFolder {
   void fold(cmpq& in, Vinstr& out) { return fold_cmp<cmpqi>(in, out); }
 
   void fold(andq& in, Vinstr& out) {
+    if (!uses[in.sf] && valid.test(in.s0) && valid.test(in.s1)) {
+      auto imm0 = vals[in.s0];
+      auto result = imm0 & vals[in.s1];
+      out = ldimmq{result, in.d};
```

Comment:
`auto const`

Also, I think this could be compacted a bit.  I'd prefer

```
auto const result = vals[in.s0] & vals[in.s1];
out = ldimmq{result, in.d};
```

or even just

```
out = ldimmq{vals[in.s0] & vals[in.s1], in.d};
```

For such a concise expression, I don't think having the extra `imm0`, e.g., is worthwhile.